packages/typst.node/src/compiler/boxed.rs

```diff
@@ -108,35 +112,28 @@ impl BoxedCompiler {
             }
         };
 
-        let Some(new_state) = new_state else {
-            return Ok(None);
-        };
-
-        world
-            .mutate_entry(new_state)
-            .map(Some)
-            .map_err(map_node_error)
+        Ok(self.universe.snapshot_with(Some(TaskInputs {
```

Comment:
I doubt that it's necessary for `snapshot_with` to have `Option<TaskInputs>` as its argument with all fields in `TaskInputs` already `Option<T>`.